lib/api/v3/users/users_api.rb

```diff
@@ -47,18 +47,20 @@ def user_transition(allowed)
               fail ::API::Errors::InvalidUserStatusTransition
             end
           end
-        end
 
-        resources :users do
-          helpers ::API::V3::Users::CreateUser
+          def authorize_user_cru_allowed
+            allowed_admin = current_user.admin? && (current_user.active? || current_user.is_a?(SystemUser))
+            allowed_user = current_user.active? && current_user.allowed_to_globally?(:add_user)
```

Comment:
That `active?` check SHOULD be superfluous. I mean that in the sense of us putting that check into the `allowed_to_globally?` method. I cannot see the consequences but I believe it should also be done for the `allowed_to?` method.

Reply:
I moved it to the UserAllowedService. The same check as in the previous line has to be made in case a `SystemUser` is passed to it.